### PR TITLE
chore: Fix a bunch of clippy warnings

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/file_encryption/attachments.rs [hunk 1]
```diff
@@ -238,10 +238,12 @@ impl<'a, R: Read + ?Sized + 'a> AttachmentEncryptor<'a, R> {
             kty: "oct".to_owned(),
             key_ops: vec!["encrypt".to_owned(), "decrypt".to_owned()],
             alg: "A256CTR".to_owned(),
+            #[allow(clippy::unnecessary_to_owned)]
             k: Base64::new(key.to_vec()),
             ext: true,
         });
-        let encoded_iv = Base64::new((iv).to_vec());
+        #[allow(clippy::unnecessary_to_owned)]
+        let encoded_iv = Base64::new(iv.to_vec());
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/file_encryption/attachments.rs:246]: Is clippy emitting false positives here?

**Review by jplatte — APPROVED** [review]

File: crates/matrix-sdk-crypto/src/file_encryption/attachments.rs (hunk 1 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/file_encryption/attachments.rs:246]: It is.

Comment (jplatte) [reply crates/matrix-sdk-crypto/src/file_encryption/attachments.rs:246]: Alright. Hadn't gotten false positives from that particular lint myself. Looks like https://github.com/rust-lang/rust-clippy/issues/9351 is tracking this.

Comment (poljar) [reply crates/matrix-sdk-crypto/src/file_encryption/attachments.rs:246]: Yeah it just started happening today. So it's pretty new.

**Pull request merged**